docs/spec/v2beta2/helmreleases.md

```diff
@@ -0,0 +1,930 @@
+applied to the ignored paths.
+
+```yaml
+spec:
+  driftDetection:
+    ignore:
+     - paths: ["/spec/replicas"]
+       target:
+         kind: Deployment
+```
+
+The following `.target` selectors are available, defining multiple fields
+causes the selector to be more specific:
+
+- `group` (Optional): Matches the `.apiVersion` group of resources while
+  offering support for regular expressions. For example, `apps`,
+  `helm.toolkit.fluxcd.io` or `.*.toolkit.fluxcd.io`.
+- `version` (Optional): Matches the `.apiVersion` version of resources while
+  offering support for regular expressions. For example, `v1`, `v2beta2` or
+  `v2beta[\d]`.
+- `kind` (Optional): Matches the `.kind` of resources while offering support
+  for regular expressions. For example, `Deployment`, `HelmRelelease` or
+  `(HelmRelease|HelmChart)`.
+- `name` (Optional): Matches the `.metadata.name` of resources while offering
+  support for regular expressions. For example, `podinfo` or `podinfo.*`.
+- `namespace` (Optional): Matches the `.metadata.namespace` of resources while
+  offering support for regular expressions. For example, `my-release-ns` or
+  `.*-system`.
+- `annotationSelector` (Optional): Matches the `.metadata.annotations` of
+  resources using a [label selector expression](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#label-selectors).
+  For example, `environment = production` or `environment notin (staging)`.
+- `labelSelector` (Optional): Matches the `.metadata.labels` of resources
+  using a [label selector expression](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#label-selectors).
+  For example, `environment = production` or `environment notin (staging)`.
+
+#### Ignore annotation
+
+To exclude certain resources from the comparison, they can be labeled or
+annotated with `helm.toolkit.fluxcd.io/driftDetection: disabled`. Using
+[post-renderers](#post-renderers), this can be applied to any resource
+rendered by Helm.
+
+```yaml
+spec:
+  postRenderers:
+    - kustomize:
+        patches:
+          - target:
+              version: v1
+              kind: Deployment
+              name: my-app
+            patch: |
+              - op: add
+                path: /metadata/annotations/helm.toolkit.fluxcd.io~1driftDetection
+                value: disabled
+```
+
+**Note:** In many cases, it may be better (and easier) to configure an [ignore
+rule](#ignore-rules) to ignore (a portion of) a resource.
+
+### Post renderers
+
+`.spec.postRenderers` is an optional list to provide [post rendering](https://helm.sh/docs/topics/advanced/#post-rendering)
+capabilities using the following built-in Kustomize directives:
+
+- [patches](https://kubectl.docs.kubernetes.io/references/kustomize/kustomization/patches/) (`kustomize.patches`)
+- [patchesStrategicMerge](https://kubectl.docs.kubernetes.io/references/kustomize/kustomization/patchesstrategicmerge/) (`kustomize.patchesStrategicMerge`)
```

Comment:
Both strategic merge and JSON patch have been deprecated and removed from KC. For consistency I think we should do the same here.

Reply:
I agree they should be deprecated but I will deal with this in a separate PR if I have time, as it is not of highest priority at the moment IMHO.

Reply:
See https://github.com/fluxcd/helm-controller/pull/832